src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTextUI.java

```diff
@@ -2598,18 +2598,21 @@ public boolean importData(JComponent comp, Transferable t) {
                     if (ic != null) {
                         ic.endComposition();
                     }
-                    Reader r = importFlavor.getReaderForText(t);
 
-                    if (modeBetween) {
-                        Caret caret = c.getCaret();
-                        if (caret instanceof DefaultCaret) {
-                            ((DefaultCaret)caret).setDot(pos, dropBias);
-                        } else {
-                            c.setCaretPosition(pos);
+                    // Use try-with-resource logic to close stream after use
+                    try (Reader r = importFlavor.getReaderForText(t)) {
+
+                        if (modeBetween) {
```

Comment:
Is this comment necessary? It's the purpose of [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html), “The try-with-resources statement ensures that each resource is closed at the end of the statement.”

I believe the blank line before the first statement in the body of the try block isn't necessary either.